### ISSUE TITLE
Measure nonlinear policy ES search-budget scaling

### ISSUE THREAD

**Issue by Ahmadfauzi34** [Ahmadfauzi34/ai_agent_builder#233]
## Purpose

Follow #231/#232. The 74-parameter `Linear(6->8)->ReLU->Linear(8->2)` policy is semantically sound and trains, but under the same OpenES budget as the 14-parameter Linear policy it finishes worse on the simple deterministic control task while costing more per rollout.

Before changing optimizer algorithms or widening model/activation surface, determine whether the comparative gap is primarily insufficient ES search budget.

## Fixed workload

Keep unchanged:

- installed Python wheel / typed host path;
- policy architecture `Linear(6->8)->ReLU->Linear(8->2)`;
- canonical parameter dimension 74;
- same four deterministic scenarios and 16-step horizon;
- identical environment dynamics, disturbances, reward/cost terms and action clamp;
- OpenES strategy 0, seed `20_260_917`, sigma 0.15, learning rate 0.06;
- Python owns environment/reward/schedule; graph owns policy execution; ES owns candidate generation/update;
- `ask_f32()` + contiguous candidate windows + `apply_flat`.

## Search-budget matrix

Use independent optimizer runs from the same zero-policy baseline:

```text
A: population 8,  generations 24  = 192 candidate evaluations (existing budget)
B: population 16, generations 24  = 384 candidate evaluations (more breadth)
C: population 8,  generations 48  = 384 candidate evaluations (more depth)
D: population 16, generations 48  = 768 candidate evaluations (breadth + depth)
```

This gives one equal-evaluation comparison (`B` vs `C`) to distinguish breadth from depth without changing objective semantics.

## Required evidence

For each case report:

- zero-policy reward;
- initial/final champion reward;
- baseline/training gain ratios;
- generation-best and champion histories;
- optimizer flags;
- candidate evaluation count;
- median ask/apply/rollout/tell timing and total training wall time;
- finite fitness and stable batch cardinality.

Across all cases preserve graph/binding identity. For the final/best selected evidence case, require exact stateful ProgramBundle parameter replay and reward replay within `1e-9`.

Selection for replay may be `argmax(final_champion_reward)` among validated cases; this is an evidence convenience, not a promoted production champion or optimizer policy.

## Interpretation

- larger budget materially improves the 74-param policy -> current issue is search scaling; do not add optimizer semantics yet until the breadth/depth pattern is understood;
- B beats C at equal eval count -> population breadth is more useful for this landscape;
- C beats B -> additional generations/depth is more useful;
- neither improves enough -> task may simply favor the smaller Linear policy, or OpenES hyperparameters/algorithm may be mismatched; investigate optimizer evidence before model-surface growth;
- runtime grows mostly with rollout count -> execution cost remains expected useful graph computation, not a reason to reopen Linear wrapper micro-optimization.

## Non-goals

No optimizer algorithm change, no ABI change, no additional activation variants, no graph/controller change, no Linear wrapper work, no zero-copy/NumPy/DLPack/PyO3, no Math Program v10, and no Go/C++ work.

Implement as a research-only slice after #232 is merged.
